kratos/processes/set_initial_state_process.cpp

```diff
@@ -0,0 +1,211 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//
+//
+
+// System includes
+
+// External includes
+
+#include "includes/model_part.h"
+#include "includes/mat_variables.h"
+
+// Project includes
+#include "processes/set_initial_state_process.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+    template<std::size_t TDim>
+    SetInitialStateProcess<TDim>::SetInitialStateProcess(
+        ModelPart& rModelPart)
+            : mrModelPart(rModelPart)
+    {
+        KRATOS_TRY
+
+        const std::size_t voigt_size = (TDim == 3) ? 6 : 3;
+
+        mInitialStrain.resize(voigt_size, false);
+        mInitialStress.resize(voigt_size, false);
+        mInitialF.resize(TDim, TDim, false);
+
+        noalias(mInitialStrain) = ZeroVector(voigt_size);
+        noalias(mInitialStress) = ZeroVector(voigt_size);
+        noalias(mInitialF)      = ZeroMatrix(TDim, TDim);
+
+        KRATOS_CATCH("")
+    }
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+    template<std::size_t TDim>
+    SetInitialStateProcess<TDim>::SetInitialStateProcess(
+        ModelPart& rModelPart,
+        const Vector& rInitialStrain,
+        const Vector& rInitialStress,
+        const Matrix& rInitialF) :
+        mrModelPart(rModelPart), mInitialStrain(rInitialStrain),
+        mInitialStress(rInitialStress), mInitialF(rInitialF)
+    {
+    }
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+    template<std::size_t TDim>
+    SetInitialStateProcess<TDim>::SetInitialStateProcess(
+        ModelPart& rModelPart,
+        const Vector& rInitialStateVector,
+        const int InitialStateType)
+            : mrModelPart(rModelPart)
+    {
+        KRATOS_TRY
+
+        const std::size_t voigt_size = (TDim == 3) ? 6 : 3;
+
+        mInitialStrain.resize(voigt_size, false);
+        mInitialStress.resize(voigt_size, false);
+        mInitialF.resize(TDim, TDim, false);
+
+        if (InitialStateType == 0) {
+            noalias(mInitialStrain) = rInitialStateVector;
+            noalias(mInitialStress) = ZeroVector(voigt_size);
```

Comment:
why do you set this? if you don't work with it you could leave it of size 0 and just check that at the moment of using it

Reply:
it can be done this way, yes. then there is an additional if in the constitutive law in order to check the size